pySDC/projects/PinTSimE/switch_estimator.py

```diff
@@ -357,3 +391,47 @@ def newton(x0, p, fprime, newton_tol, newton_maxiter):
     print(msg)
 
     return root
+
+
+class LagrangeInterpolation(object):
```

Comment:
You could consider using Thibaut's interpolation stuff, which is well tested. I am not sure how the two things compare on speed. With Thibaut's stuff, you have to generate a new interpolation matrix every time you want to interpolate to a new time and multiply this with the solution vector. But this might be faster than generating the Lagrange polynomials from scratch every time. Regardless, building on well tested stuff is always nice.